storage_controller/src/reconciler.rs

```diff
@@ -145,6 +155,18 @@ pub(crate) struct ReconcilerConfig {
 }
 
 impl ReconcilerConfig {
+    /// Configs are always constructed with an explicit priority, to force callers to think about whether
+    /// the operation they're scheduling is high-priority or not. Normal priority is not a safe default, because
+    /// scheduling something user-facing at normal priority can result in it getting starved out by background work.
+    pub(crate) fn new(priority: ReconcilerPriority) -> Self {
```

Comment:
nit: I'd move this into the builder instead.

Reply:
For now, I've made the builder also take a priority in its `new()` -- it violates the builder pattern a bit, but I really don't want to bake in a default that people get from doing `ReconcilerConfigBuilder::new().build()`, because we'll likely end up with new code paths using the wrong priority.

For ReconcilerConfig::new, I would be fine with removing it if we want to use  `ReconcilerConfigBuilder::new().build()` as a slightly more verbose alternative where we used to use ReconcilerConfig::default()